### PR TITLE
Deprecate built-in snekaers adapter

### PR DESCRIPTION
**Pull request by dixpac**
[`sneakers` gem development](https://github.com/jondot/sneakers?tab=readme-ov-file#sneakers-has-a-new-home-and-a-new-name) has been moved to [kicks gem](https://github.com/ruby-amqp/kicks)

Connected to https://github.com/rails/rails/issues/52929

### CODE REVIEW

**Comment by dixpac** [comment]
ActiveJob adapter is merged inside [sneakers/kicks gem](https://github.com/ruby-amqp/kicks/pull/12). When new gem version is released I'll finish this PR

**Comment by dixpac** [comment]
@guilleiguaran active job adapter for kicks formerly known as sneakers has been [released](https://rubygems.org/gems/kicks/versions/3.0.0).

**Comment by dhh** [comment]
I don't quite see the point of a deprecation notice when the solution simply is to switch to the new adapter. I know we talked about this a little before, @rafaelfranca, so curious to get your take. IMO, these adapters should just be yanked, and we should throw an exception telling people how to continue using them from their new location.

**Comment by dixpac** [comment]
I like removal approach. I was told [here](https://github.com/rails/rails/pull/53040) and [here](https://github.com/rails/rails/pull/52990) that deprecation is the first step.

Also, removal warnings are going to be covered in Rails release/changelog documentation.

**Comment by dhh** [comment]
Right, that's why I want to make sure we just get some input from the rest of core. It's not a hill I'm willing to die on, but looking at this, I don't see how going through deprecation first makes anything easier for anyone. But will to be overruled by more passionate voices on this topic 😄. I believe @matthewd also had some thoughts.

**Comment by matthewd** [comment]
Mostly, it's just much more disruptive to users if we force them to upgrade multiple substantial gems (here Rails + Sneakers/etc) in lock-step. It initially breaks our target flow of "upgrade Rails then look for deprecation warnings", and then when they do the multi-update, anything that _then_ breaks could be due to any of bumped gems, so it's more diagnostic work for users who are less comfortable digging through library code to figure it out.

If the built-in adapters were costing us a lot of maintainer attention regularly, I might be more inclined to accept those trade-offs to get them out sooner... but overall they've tended to need a pretty light touch. I'll for sure be glad to clean them up, but the small gain in tidiness doesn't feel like it justifies at-best-stretching our safe-upgrade promises, as well as forcing people onto very-latest gems that have potentially rushed releases to newly add the adapters. (https://github.com/ruby-amqp/kicks/pull/12#issuecomment-2433584436)

In short: it _very slightly_ softens the upgrade path for users, at an _equally slight_ level of inconvenience for us.

**Comment by dixpac** [comment]
> I'll for sure be glad to clean them up, but the small gain in tidiness doesn't feel like it justifies at-best-stretching our safe-upgrade promises, as well as forcing people onto very-latest gems that have potentially rushed releases to newly add the adapters. (https://github.com/ruby-amqp/kicks/pull/12#issuecomment-2433584436)

Thanks! I'll fix this regression in kicks gem, it was my mistake :), scoped rails dependecy from 7.2 :)

**Comment by dixpac** [comment]
I've fixed the issue in `kicks` gem, and updated version in the warning message.

**Comment by morgoth** [comment]
@dixpac Looks like this could be merged (with a small change of removal version to 8.3 or whatever will be after 8.2).
Is there anything other missing?

**Comment by dixpac** [comment]
@morgoth yes it should be good to go. Adapter is already running inside kicks for almost a year.
Let me know when do you want to merge this so I can resolve the conflicts and change deprecation warning message

**Comment by morgoth** [comment]
I'm not the maintainer so I don't have such power ;-)
Let me ask on the Rails chat

**Comment by guilleiguaran** [comment]
Please update the deprecation message and resolve the conflicts to get this merged 👍

**Comment by dixpac** [comment]
@guilleiguaran what version should I put in the deprecation message :), if this goes live in 8.2, then 9?

**Comment by zzak** [comment]
> what version should I put in the deprecation message :), if this goes live in 8.2, then 9?

Judging by the other deprecations, we should use 8.2:

```
activesupport/lib/active_support.rb:137:      "`config.active_support.to_time_preserves_timezone` is deprecated and will be removed in Rails 8.2"
activesupport/lib/active_support/configurable.rb:4:  ActiveSupport::Configurable is deprecated without replacement, and will be removed in Rails 8
.2.
activesupport/lib/active_support/core_ext/benchmark.rb:5:  active_support/core_ext/benchmark.rb is deprecated and will be removed in Rails 8.2 wit
hout replacement.
activesupport/lib/active_support/core_ext/string/multibyte.rb:39:      "String#mb_chars is deprecated and will be removed in Rails 8.2. " \
activesupport/lib/active_support/multibyte.rb:16:        "ActiveSupport::Multibyte.proxy_class= is deprecated and will be removed in Rails 8.2. "
```

I think in general we just go with the next `MINOR` and the release manager will deal with updating them all at the appropriate time, i.e. if a `MAJOR` bump is determined.

**Comment by dixpac** [comment]
@zzak ok thnx

I've updated the deprecation message.

**Comment by chaadow** [comment]
> > what version should I put in the deprecation message :), if this goes live in 8.2, then 9?
> 
> Judging by the other deprecations, we should use 8.2:
> 
> ```
> activesupport/lib/active_support.rb:137:      "`config.active_support.to_time_preserves_timezone` is deprecated and will be removed in Rails 8.2"
> activesupport/lib/active_support/configurable.rb:4:  ActiveSupport::Configurable is deprecated without replacement, and will be removed in Rails 8
> .2.
> activesupport/lib/active_support/core_ext/benchmark.rb:5:  active_support/core_ext/benchmark.rb is deprecated and will be removed in Rails 8.2 wit
> hout replacement.
> activesupport/lib/active_support/core_ext/string/multibyte.rb:39:      "String#mb_chars is deprecated and will be removed in Rails 8.2. " \
> activesupport/lib/active_support/multibyte.rb:16:        "ActiveSupport::Multibyte.proxy_class= is deprecated and will be removed in Rails 8.2. "
> ```
> 
> I think in general we just go with the next `MINOR` and the release manager will deal with updating them all at the appropriate time, i.e. if a `MAJOR` bump is determined.

@zzak But I think the release manager before the release of 8.2 will actually remove the deprecations ( from what i've seen, meaning they will :
- remove the deprecation
- Actually remove the deprecated code

I believe in this case, since it was never released, it should be `9.0`

here is an example from a recently released feature ( Immediate variants) using `9.0` for the wording: https://github.com/rails/rails/blob/69058497070e5d51315728cbd7327e8f1f039474/activestorage/app/models/active_storage/named_variant.rb#L13

**Comment by chaadow** [comment]
More examples here: https://github.com/search?q=repo%3Arails%2Frails+Rails+9.0&type=code
